### ISSUE TITLE
"No port in upstream"

### ISSUE THREAD

**Issue by rrrnld** [rvanlaar/dokku-nginx-stream#3]
I have a problem with auto-configuring proxying a TCP port. The setup looks like this:

```
$ dokku proxy:report grapevine
=====> grapevine proxy information
       Proxy enabled:                 true
       Proxy port map:                http:443:3000 http:80:3000 tcp:8008:8008
       Proxy type:                    nginx

$ cat /home/dokku/grapevine/nginx.conf
# ...
upstream grapevine-8008 {
  server 172.17.0.8:8008;
}

$ cat /etc/nginx/app-stream/grapevine.conf
server {
    listen  8008;

    proxy_pass grapevine-8008;
}
```

And I am getting the following error:

```
nginx: [emerg] no port in upstream "grapevine-8008" in /etc/nginx/app-stream/grapevine.conf:4
```

If I change the proxy_pass line to `proxy_pass 172.17.0.8:8008;` `nginx -t` stops complaining and forwarding works. What also works is manually removing the `upstream grapevine-8008` from `/home/dokku/grapevine/nginx.conf' and moving it to `/etc/nginx/app-stream/grapevine.conf`. Any idea what might cause this?

Thanks for any help, and thanks for writing this plugin!

**Comment by rrrnld** [comment]
I just saw that https://github.com/rvanlaar/dokku-nginx-stream/blob/master/templates/stream.conf.sigil should generate a config that that has the upstream definition right next to the `server` block, so I edited the file to look exactly like that.

The previous config might have been dirty, because I tried to use tcp forwarding without this plugin (just by adding it via `dokku proxy:ports-add tcp:8008:8008`. Maybe that's what broke it?
